master/internal/user/postgres_long_lived_tokens.go

```diff
@@ -0,0 +1,137 @@
+package user
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"time"
+
+	"github.com/o1egl/paseto"
+	"github.com/pkg/errors"
+	"github.com/uptrace/bun"
+
+	"github.com/determined-ai/determined/master/internal/db"
+	"github.com/determined-ai/determined/master/pkg/model"
+)
+
+const (
+	// TokenExpirationDuration is how long a newly created long lived token is valid.
+	TokenExpirationDuration = 30 * 24 * time.Hour
+)
+
+// LongLivedTokenOption is the return type for WithTokenExpiresAt helper function.
+// It takes a pointer to model.LongLivedToken and modifies it.
+// It’s used to apply optional settings to the LongLivedToken object.
+type LongLivedTokenOption func(f *model.LongLivedToken)
+
+// WithTokenExpiresAt function will add specified expiresAt (if any) to the long lived token table.
+func WithTokenExpiresAt(expiresAt *time.Time) LongLivedTokenOption {
+	return func(s *model.LongLivedToken) {
+		s.ExpiresAt = *expiresAt
+	}
+}
+
+// DeleteAndCreateLongLivedToken creates a row in the long lived token table.
+func DeleteAndCreateLongLivedToken(
+	ctx context.Context, userID model.UserID, opts ...LongLivedTokenOption,
+) (string, error) {
+	// Populate the default values in the model.
+	longLivedToken := &model.LongLivedToken{
+		UserID:    userID,
+		CreatedAt: time.Now(),
+		ExpiresAt: time.Now().Add(TokenExpirationDuration),
+	}
+
+	// Update the optional ExpiresAt field (if passed)
+	for _, opt := range opts {
+		opt(longLivedToken)
+	}
+
+	var token string
+
+	err := db.Bun().RunInTx(ctx, nil, func(ctx context.Context, tx bun.Tx) error {
+		// Tokens should have a 1:1 relationship with users, if a user creates a new token,
+		// revoke the previous token if it exists.
+		err := DeleteLongLivenTokenByUserID(ctx, userID)
```

Comment:
I could be wrong, but are the delete & insert using the transaction? It doesn't look like `tx` is being used.

Would the insert need to look something like: `tx.Bun().NewInsert()`

Reply:
You are right! Currently, the code is using db.Bun() directly, which means these calls are outside of the transaction and are not rolled back if there is an error. I will correct it.

Reply:
Done